src/generic-site/content-script.js

```diff
@@ -34,5 +35,38 @@ function updateSettings(new_settings = null) {
     }
 }
 chrome.runtime.onMessage.addListener(handleMessage);
+
+function sendRollToGameLog(message) {
+    console.log(message)
+    chrome.runtime.sendMessage(message)
+}
+
+function disconnectAllEvents() {
+    registered_events.forEach((event) => {
+        event.disconnect();
+    })
+}
+
+function handleIncomingSettings({ character, action, type} = {}) {
+    characterInternal = character;
+    chrome.runtime.sendMessage({
+        action: "hp-update",
+        character
+    })
+    chrome.runtime.sendMessage({
+        action: "settings",
+        type: "character",
+        settings: character.settings,
+        id: character.id
+    })
```

Comment:
Note that the `settings` action message is always for the global settings, never for character settings, since the character specific settings are always present in the roll request (through request.character.settings).

This is a situation I hadn't thought of though... what about character settings 🤔 Where should they be stored, what would the `id` represent (as it's not a DDB character sheet ID anymore, and we don't want conflicts/collissions).
I think some of those settings should be stored on the site's own `localStorage` if possible, rather than in Beyond20's extension settings... but if we do, we lose that info from the save/load settings option in Beyond20.

Also a site shouldn't be able to send an update to the global settings. That should already be handled on its own via the Beyond20 popup or settings alert. 

I might need to come up with a strategy to allow such sites though to tell the extension "this is a character sheet, and here are the character settings" so that when you click on the beyond20 icon in the address bar, it tells you it's a character, what its name is and lets you customize per-character settings, rather than show you the generic settings popup.

It would also be nice if the extension upon load did a search for `$(".beyond20-placeholder")` for example and injects its dice icon there for users to quick access the settings (like we do near the chat on Roll20 or Foundry, or next to the character name in DDB.